src/lib/models/errors.ts

```diff
@@ -0,0 +1,15 @@
+export class SortError extends Error {
+  error: Error;
+  errorMessage: string | undefined;
```

Comment:
Is there a case where we don't want to define the `errorMessage`?
Since we are in control of this object, we should define the `errorMessage` always, and optionally provide `error` - there might be some cases in the future when we just want to throw an error after some conditional check.

Reply:
yes